ably/params.go

```diff
@@ -0,0 +1,137 @@
+package ably
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type ProtocolType string
+
+const (
+	ProtocolJSON    = "json"
+	ProtocolMsgPack = "msgpack"
+)
+
+type Params struct {
+	RealtimeEndpoint string
+	RestEndpoint     string
+
+	ApiKey       string
+	ClientID     string
+	AppID        string
+	AppSecret    string
```

Comment:
There is no such thing as an `AppSecret`, this is actually `ApiSecret`.  I expect `AppId` is supposed to be `ApiId`